pkg/util/validation/errors.go

```diff
@@ -145,6 +145,45 @@ func newSampleTimestampTooNewError(metricName string, timestamp int64) Validatio
 	}
 }
 
+// exemplarValidationError is a ValidationError implementation suitable for exemplar validation errors.
+type exemplarValidationError struct {
+	message        string
+	seriesLabels   []cortexpb.LabelAdapter
+	exemplarLabels []cortexpb.LabelAdapter
+	timestamp      int64
+}
+
+func (e *exemplarValidationError) Error() string {
+	return fmt.Sprintf(e.message, e.timestamp, cortexpb.FromLabelAdaptersToLabels(e.seriesLabels).String(), cortexpb.FromLabelAdaptersToLabels(e.exemplarLabels).String())
+}
+
+func newExemplarEmtpyLabelsError(seriesLabels []cortexpb.LabelAdapter, exemplarLabels []cortexpb.LabelAdapter, timestamp int64) ValidationError {
+	return &exemplarValidationError{
+		message:        "exemplar missing labels:, timestamp %d series: %s labels: %s",
+		seriesLabels:   seriesLabels,
+		exemplarLabels: exemplarLabels,
+		timestamp:      timestamp,
+	}
+}
+
+func newExemplarMissingTimestampError(seriesLabels []cortexpb.LabelAdapter, exemplarLabels []cortexpb.LabelAdapter, timestamp int64) ValidationError {
+	return &exemplarValidationError{
+		message:        "exemplar missing timestamp: timestamp: %s series: %s labels: %s",
```

Comment:
Timestamp is numeric.

```suggestion
		message:        "exemplar missing timestamp, timestamp: %d series: %s labels: %s",
```